src/form/ColorMapInput.tsx

```diff
@@ -263,129 +266,126 @@ const ColorMapInput: React.FC<ColorMapProps & InjectedIntlProps> = (props) => {
+              style={{padding: "30px", flexGrow: 1, minHeight: 0,}}
+            >
+              <ColormapForm
+                currentRecord={value.customColormap.data? value.customColormap: undefined}
+                cancelAction={()=>{setShowCustomColormapModal(false)}}
+                confirmAction={(customColormap:any)=>{
+                  customColormapChanged(customColormap);
+                  setShowCustomColormapModal(false);
+                }}
+              />
+            </div>
+          </ModalBackground>
+        :null}
+        <span className={formStyles.LabelTitle}>
+          {title}
+        </span>
+        <div>
+          <div className={styles.previewColorContainer}>{colors}</div>
+          <div className={styles.MinMaxValues}>
+            <span>{minValue}</span>
+            <span>{maxValue}</span>
           </div>
-        </ModalBackground>
-        // {/* </Modal> */}
-      :null}
-      <span className={formStyles.LabelTitle}>
-        {title}
-      </span>
-      <div>
-        <div className={styles.previewColorContainer}>{colors}</div>
-        <div className={styles.MinMaxValues}>
-          <span>{minValue}</span>
-          <span>{maxValue}</span>
-        </div>
-        <div style={{position: "relative"}}>
-          <SelectBox
-            title={''}
-            choices={[["Custom colormap","Custom colormap","+ Create new colormap for this raster"],...colorMaps]}
-            value={JSON.stringify(colorMapValue.customColormap) !=="{}" && JSON.stringify(colorMapValue.options) ==="{}" ? "Custom colormap" : (colorMapType && colorMapType.colorMap) || null}
-            name={name}
-            validated={validated}
-            errorMessage={errorMessage}
-            triedToSubmit={triedToSubmit}
-            valueChanged={(colormap)=>{ 
-              colorMapChanged(colormap); 
-            }}
-            placeholder={placeholderColorMapSelection}
-            showSearchField={true}
+          <label
+            htmlFor={name}
+            className={formStyles.Label}
+          >
+            <div style={{position: "relative"}}>
+              <SelectBox
+                title={''}
+                choices={[["Custom colormap","Custom colormap","+ Create new colormap for this raster"],...colorMaps]}
+                value={JSON.stringify(colorMapValue.customColormap) !=="{}" && JSON.stringify(colorMapValue.options) ==="{}" ? "Custom colormap" : (colorMapType && colorMapType.colorMap) || null}
+                name={name}
+                validated={validated}
+                errorMessage={errorMessage}
+                triedToSubmit={triedToSubmit}
+                valueChanged={(colormap)=>{ 
+                  colorMapChanged(colormap); 
+                }}
+                placeholder={placeholderColorMapSelection}
+                showSearchField={true}
+                readOnly={readOnly}
+                form={form}
+                onFocus={onFocus}
+                onBlur={onBlur}
```

Comment:
I think another solution is to remove the `onBlur` event here. I tried it and it seems it's a good enough solution instead of moving the a part of the colormapInput state to the raster layer form.